### PR TITLE
Add new 'clean' and 'quick' parameters, correspondig to -c and -q in …

### PR DESCRIPTION
**Pull request by jasperla**
##### ISSUE TYPE

 - Feature Pull Request

##### COMPONENT NAME

`openbsd_pkg` module

##### ANSIBLE VERSION

```
ansible 2.2.0.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
```

##### SUMMARY

The OpenBSD `pkg_add` and `pkg_delete` tools have support for "purging" a package (`clean`) and all configuration files and for a "quick" installation/uninstallation that doesn't check file checksums (handy if you locally modified file and don't want to end up with a `partial-$PKGNAME`).

This PR adds two new boolean parameters to `openbsd_pkg` that allow Ansible to use the functionality described above: `clean` and `quick`.

Sample tasks:
```
    - name: Use clean
      openbsd_pkg:
        name: mpd
        state: absent
        clean: True

    - name: Use quick
      openbsd_pkg:
        name: mpd
        state: absent
        quick: True
```

@eest could you review please?

### CODE REVIEW

**Comment by eest** [comment]
Hello!

This seems reasonable to me, I have some initial thoughts:
- Instead of passing the quick and clean variables into package_present() and friends, maby just use the corresponding module.params['clean']/module.params['quick'] since we are already passing in the "module" object?
- I would add type='bool' for the "quick" and "clean" argument specification, as is done for "build", so that you cant get away with setting the variables to whatever string you want.
- In your example output above there is a "clean: True" and "quick: True" message in the result, however, I don't see this being set in the diff, was that left out?

Thanks for your interest in making the module better :).

**Comment by gregdek** [comment]
Thanks @jasperla. To the current maintainers, @eest please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit', 'needs_revision' or 'close_me' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by jasperla** [comment]
Hi @eest, thanks for your feedback. I don't quite get your last point, it's not console output but rather example tasks that I pasted.

Your first two items have been incorporated and I've also updated the examples.

File: packaging/os/openbsd_pkg.py [hunk 1]
```diff
@@ -64,6 +64,23 @@
...
+	default: no
+        description:
+          - When updating or removing packages, delete the extra configuration
+            file(s) in the old packages which are annotated with @extra in
+            the packaging-list.
+        version_added: "2.3"
+    quick:
+        required: false
+        choices: [ yes, no ]
+	default: no
```
Comment (bcoca) [inline packaging/os/openbsd_pkg.py]: bad indent/tab? same above

Comment (jasperla) [reply packaging/os/openbsd_pkg.py]: yep, bad identation.

**Comment by eest** [comment]
@jasperla: Sorry, I got confused :). What I am talking about is the output from a run of the module. It currently looks like this:

```
$ doas ansible localhost -m openbsd_pkg -a 'name=nmap state=absent clean=yes' 
 [WARNING]: Host file not found: /etc/ansible/hosts

 [WARNING]: provided hosts list is empty, only localhost is available

localhost | SUCCESS => {
    "build": false, 
    "changed": false, 
    "name": "nmap", 
    "state": "absent"
}
```

What I am thinking is if the clean/quick setting should be included there like "build" is today. This might be more of an ansible policy. @bcoca, do you have input in this?

I think the new diff looks very nice! I am currently attempting to test it using clean/quick in a validation playbook but having trouble running it because of an unrelated problem with the facts-collection code.

I have opened a ticket for that: https://github.com/ansible/ansible/issues/18473

**Comment by bcoca** [comment]
I dont think they need to be in the output, if you run with -vvv you already see the options passed in

**Comment by gregdek** [comment]
Thanks @jasperla. To the current maintainers, @eest please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit', 'needs_revision' or 'close_me' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by eest** [comment]
Currently stuck in testing because of a pkg_add problem which manifests itself when running pkg_add from a script. I have sent an e-mail to the misc@ list about the problem:
http://marc.info/?l=openbsd-misc&m=147956733622817&w=2

**Comment by eest** [comment]
I decided to go forward, patching pkg_add to temporarily fix the issue mentioned in the mail thread.
Since the problem is not something that is introduced by this I believe it is good enough to go in.

**Comment by eest** [comment]
shipit

**Comment by gregdek** [comment]
Thanks again to @jasperla for this PR, and thanks @eest for reviewing. Marking for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
Thanks @jasperla for this PR. Unfortunately, it is not mergeable in its current state due to merge conflicts. Please rebase your PR. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

For help on how to do this cleanly please see http://docs.ansible.com/ansible/community.html#contributing-code-features-or-bugfixes

[This message brought to you by your friendly Ansibull-bot.]

**Comment by jctanner** [comment]
deprecated by https://github.com/ansible/ansible/pull/19680

**Pull request closed without merging**